state/mongodb/mongodb.go

```diff
@@ -355,6 +358,11 @@ func (m *MongoDB) Query(req *state.QueryRequest) (*state.QueryResponse, error) {
 
 func getMongoURI(metadata *mongoDBMetadata) string {
 	if len(metadata.server) != 0 {
```

Comment:
Nit: there are now 4 different cases, with 4 different format strings. Perhaps we could clean up this a bit and streamline it?

Reply:
what do you have in mind? I think the variety is making the branching interesting here.